src/controls/combinators/AnyOfControl.tsx

```diff
@@ -38,31 +39,58 @@ export function AnyOfControl({
     uischemas,
   )
 
-  const combinatorSchemaSwitcher = (
-    <CombinatorSchemaSwitcher
-      config={config as unknown}
-      renderInfos={combinatorRenderInfos}
-      selectedIndex={selectedIndex}
-      setSelectedIndex={setSelectedIndex}
-      uischema={uischema}
-      path={path}
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      data={data}
-      handleChange={handleChange}
-      rootSchema={rootSchema}
-    />
-  )
+  const [renderCount, setRenderCount] = useState(1)
+
+  useEffect(() => {
+    // this is a janky workaround for two problems
```

Comment:
is the plan to work on the fix in @jsonforms/react next?

Reply:
No, I'm going to file a bug though. I'm not interested in fixing it upstream because ideally we have a different approach altogether for default creation.

Reply:
how this wasn't an issue in mercury-web?

Reply:
What we're doing in the closed-source renderer is worse:
1. we're giving the FormItem a name for a field that doesn't correspond to a property we want in the form
2. [we're setting a value for that field in a useEffect](https://github.com/greatexpectationslabs/mercury-web/blob/main/src/jsonforms/complex/AnyOfRenderer.tsx#L58-L62) so form validation doesn't complain (TBC I made this change back in December when the bug was blocking release of splitters)